sea-query 1.0.0-rc.23 fails to compile on Rust 1.92 - VALUE_SIZE exceeds 32 bytes

## Description

`sea-query 1.0.0-rc.23` fails to compile on Rust 1.92.0 with a compile-time panic about VALUE_SIZE exceeding 32 bytes. This happens even with minimal features enabled (just `sqlx-postgres`).

## Steps to Reproduce

1. Use Rust 1.92.0
2. Add `sea-orm = "2.0.0-rc.22"` (or any version that pulls in `sea-query 1.0.0-rc.23`)
3. Run `cargo build`

### Expected Behavior

The crate should compile successfully.

### Actual Behavior

```
error[E0080]: evaluation panicked: the size of Value shouldn't be greater than the expected MAX_VALUE_SIZE (32 bytes by default)
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/sea-query-1.0.0-rc.23/src/value.rs:382:31
    |
382 | pub const VALUE_SIZE: usize = check_value_size();
    |                               ^^^^^^^^^^^^^^^^^^ evaluation of \`value::VALUE_SIZE\` failed inside this call
    |
note: inside \`check_value_size\`
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/sea-query-1.0.0-rc.23/src/value.rs:419:9
    |
419 | /         panic\!(
420 | |             "the size of Value shouldn't be greater than the expected MAX_VALUE_SIZE (32 bytes by default)"
421 | |         )
    | |_________^ the failure occurred here
```

### Reproduces How Often

Always on Rust 1.92.0

## Versions

- Rust: `rustc 1.92.0 (ded5c06cf 2025-12-08)`
- sea-query: 1.0.0-rc.23
- sea-orm: 2.0.0-rc.22 (pulls in sea-query 1.0.0-rc.23)
- OS: Linux 6.18.1-arch1-2

## Additional Information

This appears to be related to changes in Rust 1.92's const evaluation or type sizes. The same code compiles fine with `sea-query 1.0.0-rc.17` (used by `sea-orm 2.0.0-rc.18`).

The workaround is to use `sea-orm 2.0.0-rc.18` which depends on `sea-query 1.0.0-rc.17`.

Related: #971, #972 (similar VALUE_SIZE issues with serde_json preserve_order)

--- a/src/value.rs
+++ b/src/value.rs
@@ -244,7 +244,6 @@ pub enum Value {
     /// so we use Arc to save space
     Enum(Option<Arc<Enum>>),
 
-    #[allow(clippy::box_collection)]
     Bytes(Option<Vec<u8>>),
 
     #[cfg(feature = "with-json")]
@@ -382,34 +381,35 @@ impl Enum {
 pub const VALUE_SIZE: usize = check_value_size();
 const MAX_VALUE_SIZE: usize = 32;
 
+#[allow(unused)]
 const EXPECTED_VALUE_SIZE: usize = {
     #[allow(unused_mut)]
     let mut max = MAX_VALUE_SIZE;
 
+    macro_rules! max_mut {
+        ($expr:expr) => {
+            let tmp = $expr;
+            if max < tmp {
+                max = tmp;
+            }
+        };
+    }
+
+    #[cfg(all(feature = "with-json", feature = "postgres-array"))]
+    // size of vec + discriminant (8)
+    max_mut!(40);
+
     // If some crate enabled indexmap feature, the size of Json will be 72 or larger.
     #[cfg(feature = "with-json")]
-    {
-        if size_of::<Option<Json>>() > max {
-            max = size_of::<Option<Json>>();
-        }
-    }
+    max_mut!(size_of::<Option<Json>>());
 
     // If bigdecimal is enabled and its size is larger, we make the limit to be bigdecimal's size
     #[cfg(feature = "with-bigdecimal")]
-    {
-        if size_of::<Option<BigDecimal>>() > MAX_VALUE_SIZE {
-            max = size_of::<Option<BigDecimal>>();
-        }
-    }
+    max_mut!(size_of::<Option<BigDecimal>>());
 
     // Jiff has extra size in debug mode. Skip size check in that case.
     #[cfg(feature = "with-jiff")]
-    {
-        let zoned_size = size_of::<Option<jiff::Zoned>>();
-        if zoned_size > max && cfg!(debug_assertions) {
-            max = zoned_size;
-        }
-    }
+    max_mut!(size_of::<Option<jiff::Zoned>>());
 
     max
 };

